docs/calling-downstream-apis/custom-apis.md

```diff
@@ -0,0 +1,823 @@
+
+| Approach | Complexity | Flexibility | Use Case |
+|----------|------------|-------------|----------|
+| **IDownstreamApi** | Low | Medium | Standard REST APIs with configuration |
+| **MicrosoftIdentityMessageHandler** | Medium | High | HttpClient with DI and composable pipeline |
+| **IAuthorizationHeaderProvider** | High | Very High | Complete control over HTTP requests |
+
+## IDownstreamApi - Recommended for Most Scenarios
+
+`IDownstreamApi` provides a simple, configuration-driven approach for calling REST APIs with automatic token acquisition.
+
+### Installation
+
+```bash
+dotnet add package Microsoft.Identity.Web.DownstreamApi
+```
+
+### Configuration
+
+Define your API in appsettings.json:
+
+```json
+{
+  "AzureAd": {
+    "Instance": "https://login.microsoftonline.com/",
+    "TenantId": "your-tenant-id",
+    "ClientId": "your-client-id",
+    "ClientCredentials": [
+      {
+        "SourceType": "ClientSecret",
+        "ClientSecret": "your-client-secret"
+      }
+    ]
+  },
+  "DownstreamApis": {
+    "MyApi": {
+      "BaseUrl": "https://api.example.com",
+      "Scopes": ["api://my-api-client-id/read", "api://my-api-client-id/write"],
+      "RelativePath": "api/v1",
+      "RequestAppToken": false
+    },
+    "PartnerApi": {
+      "BaseUrl": "https://partner.example.com",
+      "Scopes": ["api://partner-api-id/.default"],
+      "RequestAppToken": true
+    }
+  }
+}
+```
+
+### ASP.NET Core Setup
+
+```csharp
+using Microsoft.Identity.Web;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add authentication
+builder.Services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
+    .AddMicrosoftIdentityWebApp(builder.Configuration.GetSection("AzureAd"))
+    .EnableTokenAcquisitionToCallDownstreamApi()
+    .AddInMemoryTokenCaches();
+
+// Register downstream APIs
+builder.Services.AddDownstreamApi("MyApi",
+    builder.Configuration.GetSection("DownstreamApis:MyApi"));
+    
+builder.Services.AddDownstreamApi("PartnerApi",
+    builder.Configuration.GetSection("DownstreamApis:PartnerApi"));
+
+builder.Services.AddControllersWithViews();
+
+var app = builder.Build();
+app.UseAuthentication();
+app.UseAuthorization();
+app.MapControllers();
+app.Run();
+```
+
+### Basic Usage
+
+```csharp
+using Microsoft.Identity.Abstractions;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+[Authorize]
+public class ProductsController : Controller
+{
+    private readonly IDownstreamApi _api;
+    
+    public ProductsController(IDownstreamApi api)
+    {
+        _api = api;
+    }
+    
+    // GET request
+    public async Task<IActionResult> Index()
+    {
+        var products = await _api.GetForUserAsync<List<Product>>(
+            "MyApi",
+            "products");
+        
+        return View(products);
+    }
+    
+    // GET with route parameters
+    public async Task<IActionResult> Details(int id)
```

Comment:
Is this correct? Should there not be a route with a template to specify where id comes from?